src/main/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotChangeEventSource.java

```diff
@@ -480,7 +482,16 @@ protected SnapshotResult<YugabyteDBOffsetContext> doExecute(ChangeEventSourceCon
                   if (!snapshotCompletedTablets.contains(part.getId()) && taskContext.shouldEnableExplicitCheckpointing()) {
                     doSnapshotCompletionCheck(part, snapshotCompletedTablets, tabletsWaitingForCallback, previousOffset);
                   }
-                  continue;
+
+                  if (!snapshotCompletedTablets.contains(part.getId()) && isCallbackTimeoutExceeded(part)) {
```

Comment:
Do we handle empty tablets?

Reply:
Yes, for empty tablets this flow will never be hit as `snapshotCompletedTablets` will contain the partition and the condition `!snapshotCompletedTablets.contains(part.getId())` will become `false`.